tcms/testplans/tests/test_views.py

```diff
@@ -54,3 +55,125 @@ def test_view_tags_without_permissions(self):
+            'product': cls.product.pk,
+            'product_version': cls.product_version.pk,
+            'type': cls.test_plan_type.pk,
+            'name': cls.test_plan_1.name,
+
+            'email_settings-0-auto_to_plan_author': 'off',
+            'email_settings-0-auto_to_case_owner': 'off',
+            'email_settings-0-auto_to_case_default_tester': 'off',
+            'email_settings-0-notify_on_case_update': 'off',
+            'email_settings-0-notify_on_plan_update': 'off',
+
+            'email_settings-0-id': cls.test_plan_1.emailing.pk,
+            'email_settings-TOTAL_FORMS': '1',
+            'email_settings-INITIAL_FORMS': '1',
+            'email_settings-MIN_NUM_FORMS': '0',
+            'email_settings-MAX_NUM_FORMS': '1',
+            'is_active': True,
+        }
+
+    def test_show_testplan_edit_page(self):
+        response = self.client.get(self.testplan_edit_url)
+        self.assertContains(response, _('Edit TestPlan'))
+        self.assert_notify_form(response)
+        self.assertContains(
+            response,
+            '<input class="bootstrap-switch" name="is_active" type="checkbox" checked>',
+            html=True)
+
+    def test_edit_testplan_text_field(self):
+        edit_data = self.testplan_edit_data.copy()
+        new_text = 'Edited: {0}'.format(self.test_plan_1.text)
+        edit_data['text'] = new_text
+
+        response = self.client.post(self.testplan_edit_url, data=edit_data, follow=True)
+
+        self.assertContains(response, new_text, html=True)
+        self.test_plan_1.refresh_from_db()
+        self.assertEqual(new_text, self.test_plan_1.text)
+
+    # Issue: When there is an invalid value on notify_formset,
+    # returned response is always the default (ON).
+    def test_with_invalid_notify_form_value(self):
+        test_plan = TestPlanFactory(
+            author=self.tester)
+
+        testplan_edit_url = reverse('plan-edit', args=[test_plan.pk])
+        edit_data = self.testplan_edit_data.copy()
+        edit_data['email_settings-0-auto_to_plan_author'] = 'invalid_value'
+
+        response = self.client.post(testplan_edit_url, data=edit_data, follow=True)
+
+        self.assert_notify_form(response)
+
+    def test_with_invalid_product_shows_error(self):
+        edit_data = self.testplan_edit_data.copy()
+        edit_data['product'] = -1
+
+        response = self.client.post(self.testplan_edit_url, data=edit_data, follow=True)
+
+        self.assertContains(
+            response,
+            _('Select a valid choice. That choice is not one of the available choices.'))
+
+    def test_with_invalid_type_shows_error(self):
+        edit_data = self.testplan_edit_data.copy()
+        edit_data['type'] = -1
+
+        response = self.client.post(self.testplan_edit_url, data=edit_data, follow=True)
+
+        self.assertContains(
+            response,
+            _('Select a valid choice. That choice is not one of the available choices.'))
```

Comment:
This scenario must verify that all of the on/off buttons are as we expect.

To make sure the currently selected value has been kept between errors (e.g. that the patch works) one of the buttons in notify form should be set to a different value than its default.